syntax = "proto3";

package FaceIdenty;

service FaceIdenty {
	rpc HelloFace(HelloFaceReq) returns(HelloFaceRsp);
	rpc InitFace(InitFaceReq) returns(InitFaceRsp);
	rpc Detect(DetectReq)returns(PredictRsp);
	rpc FindSimilarFace(FindSimilarFaceReq) returns (FindSimilarFaceRsp);
	// rpc FindSimilarHistoryFace(FindSimilarFaceReq) returns (FindSimilarFaceRsp);
	rpc FindSimilarHistoryFace(FindSimilarFaceReq) returns (PredictRsp);
	rpc Compare(CompareReq)returns (CompareRsp);
	rpc Predict(PredictReq)returns(PredictRsp);
}
////////////////////////////////////////////////////////////////////////////////////////
message DetectReq{
	string id = 1;
	uint32 width = 2;
	uint32 height = 3; 
	bytes pic = 4;
}

////////////////////////////////////////////////////////////////////////////////////////
message PredictFaceInfo{
	string faceid = 1;
	uint32 left = 2;
	uint32 top = 3;
	uint32 width = 4;
	uint32 height = 5;
	string personid =6;
	bool  newperson = 7;
	float confidence = 8;
}
message PredictReq{
	string id = 1;
	uint32 width = 2;
	uint32 height = 3; 
	bytes pic = 4;
}

message PredictRsp{
	string id = 1;
	repeated PredictFaceInfo predictFaces = 2;
}

////////////////////////////////////////////////////////////////////////////////////////
message CompareReq{
	repeated  faceInfo faces = 1;
}
message CompareRsp{
	string compareInfo = 1;
}
////////////////////////////////////////////////////////////////////////////////////////
message InitFaceReq{
	string peerId = 1;
}
message InitFaceRsp{
	int32  error = 1;
	string  error_message = 2;
}
////////////////////////////////////////////////////////////////////////////////////////
message faceInfo{
	string Id = 1;
	bytes facePic = 2;
}

message verifyInfo{
	string Id = 1;
	string similarityId = 2;
	float  score = 3;
}

message FindSimilarFaceReq{
	float    threshold = 1;
	string   peerId = 2;
	int64    timestamp = 3;
	repeated  faceInfo faces = 4;
}

message FindSimilarFaceRsp{
	repeated verifyInfo verifys = 1;
}


////////////////////////////////////////////////////////////////////////////////////////
message HelloFaceReq{
	string  ask = 1;
}
message HelloFaceRsp{
	string response = 1;
}

